plutus-core/plutus-core/src/PlutusCore/Builtin/TestKnown.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE FlexibleInstances       #-}
+{-# LANGUAGE MultiParamTypeClasses   #-}
+{-# LANGUAGE QuantifiedConstraints   #-}
+{-# LANGUAGE TypeFamilies            #-}
+{-# LANGUAGE TypeOperators           #-}
+{-# LANGUAGE UndecidableInstances    #-}
+{-# LANGUAGE UndecidableSuperClasses #-}
+
+module PlutusCore.Builtin.TestKnown
```

Comment:
Created a separate file just not to keep this scary looking stuff in the important `KnownType` module and in order not to introduce a dependency on `KnownTypeAst` in it, as the two concepts are explicitly orthogonal (described in a Note).

Reply:
Why not put this rationale in the module comment?

Reply:
Good point.